lib/rom/sql/relation.rb

```diff
@@ -39,6 +39,26 @@ def project(*names)
       end
 
       # @api public
+      def group(*args, &block)
+        __new__(dataset.__send__(__method__, *args, &block))
+      end
+
+      # @api public
+      def group_and_count(*args, &block)
+        __new__(dataset.__send__(__method__, *args, &block))
+      end
+
+      # @api public
+      def select_append(*args, &block)
+        __new__(dataset.__send__(__method__, *args, &block))
+      end
+
+      # @api public
+      def select_group(*args, &block)
+        __new__(dataset.__send__(__method__, *args, &block))
+      end
```

Comment:
we could dry it up but I'm not sure if it's worth it, those methods will not change at all. As @mcls pointed out it's just to define the interface explicitly and document it.